src/prefect/cli/deployment.py

```diff
@@ -557,7 +557,7 @@ async def run(
             multi_params = json.loads(multiparams)
         except ValueError as exc:
             exit_with_error(f"Failed to parse JSON: {exc}")
-        if not watch and watch_interval != 5:
+        if watch_interval and not watch:
```

Comment:
ah my bad, I didn't realize we have a default `watch_interval`! so I don't think that this is needed.

Reply:
I was back and forth on this and was going to just print a warning, but honestly this may prevent someone from accidentally not getting their planned behavior, so I'd like to leave it in.

Reply:
oh I mean `if watch_interval` is always going to evaluate to true if we've given it a default though. so I think this would just error if users don't provide `watch`.

Reply:
nice! yeah I think forcing users to be explicit is definitely preferable to users thinking their flow run is being watched when it isn't